app/main/views/index.py

```diff
@@ -32,6 +37,13 @@ def download_document_download(service_id, document_id):
     if not key:
         abort(404)
 
+    user_id = session.get('user_id')
+
+    # the user hasn't clicked through the landing page - redirect them back there
+    if current_user.get_id() != user_id:
+        url = url_for('main.download_document_landing', service_id=service_id, document_id=document_id, key=key)
+        return redirect(url)
+
```

Comment:
At the moment `download_link` below exposes the internal cloudapps domain to the user, which I don't think we want to do

Reply:
Yep I agree there should be a story but if not I'll add one